tests/native_api_tests/test_api_model.cpp

```diff
@@ -351,37 +375,183 @@ TEST_CASE("API Model") {
+
+        // input dataset - col
+        DatasetConst input_dataset_col{"input", 0, 1};
+        input_dataset_col.add_buffer("node", 1, 1, nullptr, nullptr);
+        input_dataset_col.add_attribute_buffer("node", "id", node_id.data());
+        input_dataset_col.add_attribute_buffer("node", "u_rated", node_u_rated.data());
+
+        input_dataset_col.add_buffer("source", 1, 1, nullptr, nullptr);
+        input_dataset_col.add_attribute_buffer("source", "id", input_source_id.data());
+        input_dataset_col.add_attribute_buffer("source", "node", input_source_node.data());
+        input_dataset_col.add_attribute_buffer("source", "status", input_source_status.data());
+        input_dataset_col.add_attribute_buffer("source", "u_ref", input_source_u_ref.data());
+        input_dataset_col.add_attribute_buffer("source", "sk", input_source_sk.data());
+        input_dataset_col.add_attribute_buffer("source", "rx_ratio", input_source_rx_ratio.data());
+
+        input_dataset_col.add_buffer("sym_load", 1, 1, nullptr, nullptr);
+        input_dataset_col.add_attribute_buffer("sym_load", "id", input_sym_load_id.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "node", input_sym_load_node.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "status", input_sym_load_status.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "type", input_sym_load_type.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "p_specified", input_sym_load_p_specified.data());
+        input_dataset_col.add_attribute_buffer("sym_load", "q_specified", input_sym_load_q_specified.data());
+
+        // update dataset
+        std::vector<Idx> input_source_indptr{0, 1, 1};
+        std::vector<ID> const update_source_id{1};
+        std::vector<double> const update_source_u_ref{0.5};
+        Buffer update_source_buffer{PGM_def_update_source, 1};
+        update_source_buffer.set_nan();
+        update_source_buffer.set_value(PGM_def_update_source_id, update_source_id.data(), -1);
+        update_source_buffer.set_value(PGM_def_update_source_u_ref, update_source_u_ref.data(), -1);
+
+        std::vector<Idx> input_sym_load_indptr{0, 1, 2};
+        std::vector<ID> const update_sym_load_id{2, 5};
+        std::vector<double> const update_sym_load_q_specified{100.0, 300.0};
+        Buffer update_sym_load_buffer{PGM_def_update_sym_load, 2};
+        update_sym_load_buffer.set_nan();
+        update_sym_load_buffer.set_value(PGM_def_update_sym_load_id, update_sym_load_id.data(), -1);
+        update_sym_load_buffer.set_value(PGM_def_update_sym_load_q_specified, update_sym_load_q_specified.data(), -1);
+
+        // update dataset - row
+        DatasetConst update_dataset_row{"update", 1, 2};
+        update_dataset_row.add_buffer("source", -1, 1, input_source_indptr.data(), update_source_buffer);
+        update_dataset_row.add_buffer("sym_load", -1, 2, input_sym_load_indptr.data(), update_sym_load_buffer);
```

Comment:
maybe you can also add one that is dense instead of sparse (or make one of the two dense)

Reply:
Will be added via the row PR.